python/ray/utils.py

```diff
@@ -489,6 +490,15 @@ def get_system_memory():
     return psutil_memory_in_bytes
 
 
+def get_num_cpus():
+    try:
+        return int(subprocess.check_output("nproc"))
```

Comment:
consider also allowing this behavior to be overriden with an env var, like `RAY_DETECT_NPROC=0`

Reply:
I am on the same page we should probably have more warning msgs in case this change broke things silenty, but it doesn’t make sense to me to have an option to turn it off. Is it for isolating user issues in the future?

Reply:
Think about it like this - effectively, `multiprocessing.cpu_count()` is "broken". But by pushing this change, in a vague sense, Ray becomes "incompatible with multiprocessing, psutil, and os.cpu_count". It is also likely that the world revolves around multiprocessing and its behaviors, rather than Ray.
 
Also, think about the person who is developing a library on top of Modin/or some other library. They do not even know that ray even exists. Suddenly their entire codebase breaks - should they pin ray to older version? should they stick "ray.init" everywhere in their code?

Reply:
Btw, libraries which break, under these conditions also break if they're run on bare metal (or a vm) with 1 cpu. It should probably be a best practice for libraries to check that they have enough CPU's available. Modin, for example, would not have to do this because it works with 1 CPU, whereas many of our libraries would not.